### PR TITLE
add async support from POST /Deploy issue #1802

### PR DESCRIPTION
**Pull request by suwatch**

### CODE REVIEW

File: Kudu.FunctionalTests/Vfs/VfsControllerBaseTest.cs [hunk 1]
```diff
@@ -28,7 +29,7 @@ public class VfsControllerBaseTest
 
         private static readonly ContentRangeHeaderValue _fileContentRange = new ContentRangeHeaderValue(_fileContent0.Length);
 
-        private static readonly string _conflict = "<<<<<<< HEAD\r\nAAAA\r\nbbb\r\nCCCCC\r\n=======\r\nCCCCC\r\nbbb\r\nAAAAA\r\n>>>>>>>";
+        private static readonly string _conflict = "^<<<<<<< .*\r\nAAAA\r\nbbb\r\nCCCCC\r\n=======\r\nCCCCC\r\nbbb\r\nAAAAA\r\n>>>>>>>";
```
Comment (suwatch) [inline Kudu.FunctionalTests/Vfs/VfsControllerBaseTest.cs]: This is unrelated fix.  It addressed git 2.6.3 message changed.

File: Kudu.Services/ServiceHookHandlers/FetchHandler.cs [hunk 2]
```diff
@@ -129,15 +129,34 @@ public class FetchHandler : HttpTaskAsyncHandler
...
+                                                            out tempChangeSet,
+                                                            deployInfo.TargetChangeset,
+                                                            deployInfo.Deployer);
+                        }
+
+                        PerformBackgroundDeployment(deployInfo, _environment, _settings, _tracer.TraceLevel, context.Request.Url, tempDeployment, tempChangeSet);
                     }
 
+                    // latest deployment keyword reserved to poll till deployment done
+                    context.Response.Headers["Location"] = new Uri(context.Request.Url,
```
Comment (davidebbo) [inline Kudu.Services/ServiceHookHandlers/FetchHandler.cs]: Only if async

File: Kudu.Services/Deployment/DeploymentController.cs [hunk 3]
```diff
@@ -392,6 +399,37 @@ public HttpResponseMessage GetResult(string id)
...
+                {
+                    var results = _deploymentManager.GetResults();
+                    var pending = results.Where(r => r.Status != DeployStatus.Success && r.Status != DeployStatus.Failed).FirstOrDefault();
+                    if (pending != null)
+                    {
+                        _tracer.Trace("Deployment {0} is {1}", pending.Id, pending.Status);
+                        return true;
+                    }
+
+                    var latest = results.Where(r => r.EndTime != null).OrderBy(r => r.EndTime.Value).LastOrDefault();
```
Comment (shrimpy) [inline Kudu.Services/Deployment/DeploymentController.cs]: is the EndTime null check necessary?
after above pending deployment check, there shouldn`t be any deployment that without "EndTime" right?
